doc/Language/rb-nutshell.pod

```diff
@@ -0,0 +1,944 @@
+
+=head3 C«<=> leg» Three-way comparisons
+
+In Ruby, the C«<=>» operator returns -1, 0, or 1.
+In Perl 6, they return C<Order::Less>, C<Order::Same>, or C<Order::More>.
+
+C«<=>» forces numeric context for the comparison.
+
+C«leg» ("Less, Equal, or Greater?") forces string context for the comparison.
+
+C«cmp» does either C«<=>» or C<leg>, depending on the existing type of its
+arguments.
+
+=head3 C<~~> Smart-match operator
+
+This is a very common matching operator. Here are some examples:
+
+    say "match!" if $foo ~~ /bar/;      # Regex match
+    say "match!" if $foo ~~ "bar";      # String match
+    say "match!" if $foo ~~ :(Int, Str) # Signature match (destructure)
+
+See L<S03/Smart matching|https://design.perl6.org/S03.html#Smart_matching>
+
+=head3 C<& | ^> Numeric Bitwise ops
+=head3 C<& | ^> Boolean ops
+
+In Perl 6, these single-character ops have been removed, and replaced by
+two-character ops which coerce their arguments to the needed context.
+
+    # Infix ops (two arguments; one on each side of the op)
+    +&  +|  +^  And Or Xor: Numeric
+    ~&  ~|  ~^  And Or Xor: String
+    ?&  ?|  ?^  And Or Xor: Boolean
+
+    # Prefix ops (one argument, after the op)
+    +^  Not: Numeric
+    ~^  Not: String
+    ?^  Not: Boolean (same as the ! op)
+
+=head3 C<&.> Conditional chaining operator
+
+Ruby uses the C<&.> operator to chain methods without raising an error if one
+invocation returns nil. In Perl 6 use C<.?> for that.
+
+=head3 C«<< >>» Numeric shift left|right ops
+
+Replaced by C«+<» and C«+>» .
+
+    puts 42 << 3  # Ruby
+    say  42 +< 3; # Perl 6
+
+=head3 C«=>» and C<:> Key-Value Separators
+
+In Ruby, C«=>» is used in the context of key/value pairs for Hash literal
+declaration and parameter passing. C<:> is used as a shorthand when the left
+side is a symbol.
+
+In Perl 6, C«=>» is the Pair operator, which is quite different in
+principle, but works the same in many situations.
+
+If you were using C«=>» in hash initialization, or in passing arguments to a
+method that expects named values or a hash, then the usage is very similar.
+
+    # Works similarly in Ruby and Perl 6
+    my %hash = ( AAA => 1, BBB => 2 );
+    get_the_loot( 'diamonds', { quiet_level => 'very', quantity => 9 }); # Note the curly braces
+
+=head3 C<? :> Ternary operator
+
+Now spelled with two question marks instead of one question mark, and two
+exclamation points instead of one colon.
+
+    result     = (  score > 60 )  ? 'Pass'  : 'Fail'; # Ruby
+    my $result = ( $score > 60 ) ?? 'Pass' !! 'Fail'; # Perl 6
+
+=head3 C<+> String Concatenation
+
+Replaced by the tilde.
+
+Mnemonic: think of "stitching" together the two strings with needle and thread.
+
+    $food = 'grape' + 'fruit'  # Ruby
+    $food = 'grape' ~ 'fruit'; # Perl 6
+
+=head3 String interpolation
+
+In Ruby, C<"#{foo}s"> deliminates an arbitrary block embedded in the string.
+In Perl 6 drop the C<#> prefix: C<"{$foo}s">
```

Comment:
As in Ruby, you can place arbitrary code into the embedded block and it will be rendered in string context.

Ruby uses .to_s to get string context, P6 uses .Str . Or does it use .gist or .perl or something for this?